src/app/shared/services/http-action-trigger.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core'
+import { catchError, Observable } from 'rxjs'
+import { ActionTriggerService } from '../abstractions/action-trigger-service'
+import { ActionTrigger } from '../models/action-trigger'
+import { HttpClient } from '@angular/common/http'
+import { HttpErrorService } from '../../core/services/http-error.service'
+import { environment } from '../../../environments/environment'
+
+const ACTION_TRIGGER_URL: string = `${environment.apiBaseUrl}/actionTriggers`
+
+@Injectable()
+export class HttpActionTriggerService extends ActionTriggerService {
+  constructor(
+    private readonly http: HttpClient,
+    private readonly httpErrorService: HttpErrorService
+  ) {
+    super()
+  }
+
+  public getActionTriggers(): Observable<ActionTrigger[]> {
+    return this.http.get<ActionTrigger[]>(ACTION_TRIGGER_URL).pipe(catchError(error => this.httpErrorService.catchError(error)))
+  }
```

Comment:
We say that what we get is an array of ActionTrigger, but the result is not validated. to ensure that the result is actually of this type, like we do for other entities/actions.

Reply:
The data is now being validated.